state/futures/actions.ts

```diff
@@ -992,11 +994,17 @@ export const modifyIsolatedPosition = createAsyncThunk<
 >(
 	'futures/modifyIsolatedPosition',
 	async ({ sizeDelta, delayed, offchain }, { getState, dispatch, extra: { sdk } }) => {
+		const account = selectFuturesAccount(getState());
 		const marketInfo = selectMarketInfo(getState());
-		const priceImpact = selectPriceImpactOrDesiredFill(getState());
+		const priceImpact = selectIsolatedPriceImpact(getState());
+		const desiredFill = selectDesiredTradeFillPrice(getState());
 		if (!marketInfo) throw new Error('Market info not found');
+		if (!account) throw new Error('Account not connected');
 
 		try {
+			const isFlagged = await sdk.futures.getIsFlagged(account, marketInfo.market);
+			const priceImpactOrFill = !!isFlagged ? desiredFill : priceImpact;
```

Comment:
Not sure this will work will it, as isFlagged only returns when the account is flagged for liquidation, so this would return false when not reverting? We should prob make a special temporary function, `isFlaggedReverts` or something and return true on revert.

Reply:
You're right - I thought I committed `isFlagged == null` since that will work here and that's actually the version I tested.

Reply:
The function sends `null` if the transaction reverts in the sdk, since I added a `.catch()`